### PR TITLE
chore(javascript): use oxlint

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket:

### Changes included:

trying out oxlint instead of oslint, performance improvements on the ci are considerable, the output changes but nothing crazy, mostly syntactic sugar

let's discuss it

|step|eslint|oxlint|
|----|------|------|
|gen|2m50|50s|
|cts|1m50|41s|
|snippets|50s|37s|
|guides|21s|10s|

### CODE REVIEW

**Comment by millotp** [comment]
ahah I tried it too last week, turns out its not a formatter and doesn't remove white spaces or any stylistic fixes (https://oxc.rs/docs/guide/usage/linter#language-support), so not ideal for generated code

File: clients/algoliasearch-client-javascript/nx.json [hunk 1]
```diff
@@ -3,11 +3,7 @@
     "default": {
       "runner": "nx/tasks-runners/default",
       "options": {
-        "cacheableOperations": [
-          "build:many",
-          "build:all",
-          "build"
-        ]
+        "cacheableOperations": ["build:many", "build:all", "build"]
```
Comment (millotp) [inline clients/algoliasearch-client-javascript/nx.json:6]: what changed for json file ? they should still use eslint right ?

**Review by millotp — COMMENTED** [review]
can you format the playground too pls ? I think there is a bunch of changes

File: clients/algoliasearch-client-javascript/nx.json (hunk 1 above)
Comment (shortcuts) [reply clients/algoliasearch-client-javascript/nx.json:6]: good catch, I forgot to add it back after my tests https://github.com/algolia/api-clients-automation/pull/3895/commits/b390a8b3ff979490d31c3d942c3469e9512937b6

**Review by millotp — APPROVED** [review]
lezzzgo, bye bye eslint u 2 slow

**Pull request merged**